erazhu31/TLS Notary DOracle/smart-contract/daemon/daemon.ts

```diff
@@ -0,0 +1,145 @@
+		}
+	}
+
+	async checkIdentity(identity: string, candidate: string, purpose: number): Promise<boolean>
+	{
+		try
+		{
+			return identity == candidate || await (new ethers.Contract(identity, IERC734.abi, this.wallet.provider)).keyHasPurpose(utils.addrToKey(candidate), purpose);
+		}
+		catch
+		{
+			console.log(identity, candidate)
+			return false;
+		}
+	}
+
+	async getVerifiedResult(doracleCallId: string) : Promise<string>
+	{
+			let task = await this.iexechub.viewTask(doracleCallId);
+			let deal = await this.iexecclerk.viewDeal(task.dealid);
+
+			if (this.requester)
+			{
+				utils.require(deal.requester == this.requester, "requester filtered (this is not an error)");
+			}
+
+			utils.require(task.status == 3, "result-not-available");
+			utils.require(task.resultDigest == ethers.utils.keccak256(task.results), "result-not-validated-by-consensus");
+			utils.require(this.settings.authorizedApp        == ethers.constants.AddressZero || await this.checkIdentity(this.settings.authorizedApp,        deal.app.pointer,        this.settings.GROUPMEMBER_PURPOSE), "unauthorized-app");
+			utils.require(this.settings.authorizedDataset    == ethers.constants.AddressZero || await this.checkIdentity(this.settings.authorizedDataset,    deal.dataset.pointer,    this.settings.GROUPMEMBER_PURPOSE), "unauthorized-dataset");
+			utils.require(this.settings.authorizedWorkerpool == ethers.constants.AddressZero || await this.checkIdentity(this.settings.authorizedWorkerpool, deal.workerpool.pointer, this.settings.GROUPMEMBER_PURPOSE), "unauthorized-workerpool");
+			utils.require(this.settings.requiredtrust <= deal.trust, "invalid-trust");
+
+			// Check tag - must be done byte by byte.
+			let [ ta, rta ] = [ deal.tag, this.settings.requiredtag ].map(ethers.utils.arrayify);
+			for (var i in ta) utils.require((rta[i] & ~ta[i]) == 0, "invalid-tag");
+
+			return task.results;
+	}
+
+	async checkData(data: string) : Promise<void>
+	{
+		let [ date, details, value ] = ethers.utils.defaultAbiCoder.decode(["uint256", "string", "uint256"], data);
+		let entry = await this.doracle.values(ethers.utils.solidityKeccak256(["string"],[details]));
+		utils.require(entry.date < date, "new-value-is-too-old");
+	}
+
+	trigger(doracleCallId: string, event: {})
+	{
+		process.stdout.write(`${new Date().toISOString()} | processing ${doracleCallId} ... `);
+		this.getVerifiedResult(doracleCallId)
+		.then(data => {
+			this.checkData(data)
+			.then(() => {
+				this.doracle.processResult(doracleCallId)
+				.then(tx => {
+					process.stdout.write(`success\n`);
+				})
+				.catch(e => {
+					const txHash = e.transactionHash;
+					const data   = e.data[txHash];
+					process.stdout.write(`Error: ${data.error} (${data.reason})\n`);
+				});
+			})
+			.catch(reason => {
+				process.stdout.write(`Invalid results (${reason})\n`);
+			});
+		})
+		.catch(reason => {
+			process.stdout.write(`Failled to verify results (${reason})\n`);
+		});
+	}
+}
```

Comment:
This file makes sens for the pricefeed, but is completelly useless here. Either update it to you oracle or remove it!